### ISSUE TITLE
ErrorIfDeprecated when 'gpt_sync_mbr' is used

### ISSUE THREAD

**Issue by jsmeix** [rear/rear#3148]
Support for the SUSE specific 'gpt_sync_mbr' partitioning scheme
(see https://github.com/rear/rear/issues/544 what that is)
should be dropped in ReaR because
the SUSE specific patches for 'gpt_sync_mbr' in parted were
removed in March 2016 for SLES12 SP2 so 'gpt_sync_mbr'
is no longer supported by SUSE since 2016.

This issue is triggered by
https://github.com/rear/rear/pull/3145
therein in particular
https://github.com/rear/rear/pull/3145#discussion_r1481388431
that reads (excerpt)
```
I think you can safely ignore the 'gpt_sync_mbr' partitioning.

Details:

I found https://build.opensuse.org/request/show/519107
where the SUSE specific patches to parted were removed
that implemented the 'gpt_sync_mbr' partitioning.
The comment therein (excerpt)
---------------------------------------------------------------
clean-up regarding pMBR handling to sync with SLE12-SP*
---------------------------------------------------------------
indicates that the 'gpt_sync_mbr' partitioning
was no longer supported since some SLES12 service pack.

Further investigation with the openSUSE build service
tool 'osc' shows
---------------------------------------------------------------
# osc search parted | grep '^SUSE:SLE-12'
SUSE:SLE-12-SP1:GA                       parted
SUSE:SLE-12-SP1:Update                   parted
SUSE:SLE-12-SP2:GA                       parted
SUSE:SLE-12-SP2:Update                   parted
SUSE:SLE-12-SP3:GA                       parted
SUSE:SLE-12-SP3:Update                   parted
SUSE:SLE-12-SP4:GA                       parted
SUSE:SLE-12-SP4:Update                   parted
SUSE:SLE-12:GA                           parted

# osc cat SUSE:SLE-12-SP2:GA parted parted.changes
...
Thu Mar 24 11:52:51 UTC 2016 - puzel@suse.com

- Drop (SUSE specific) support for hybrid pMBR (gpt_sync_mbr
  label) (fate#317849)
  - remove: parted-gpt-mbr-sync.patch
  - remove: libparted-ppc-prepboot-in-syncmbr.patch
  - refresh patches
---------------------------------------------------------------

SUSE:SLE-12-SP1:GA parted parted.changes does not contain that.

So the SUSE specific patches for 'gpt_sync_mbr' were removed
for SLES12 SP2.

This means a SLES12 system that was installed with SLES12
before SLES12 SP2 could have 'gpt_sync_mbr' partitioning.
Subsequent system upgrades to further service packs only
upgrade RPM packages but leave the partitioning unchanged.
I think that one can also upgrade from SLES12 to SLES15
without new partitioning (i.e. without installing from scratch)
so there could be even SLES15 systems with old inherited
'gpt_sync_mbr' partitioning.

If issues appear in ReaR because of an old inherited
'gpt_sync_mbr' partitioning, it is SUSE's problem
```

Because there could be recent SLES15 systems
with old inherited 'gpt_sync_mbr' partitioning
users with such systems should get an obvious information
that 'gpt_sync_mbr' support is deprecated in ReaR
so we at ReaR upstream cas see if there are really
users with old inherited 'gpt_sync_mbr' partitioning.

In particular I never again tested a system
with 'gpt_sync_mbr' partitioning since the time
when I had implemented basic support for it
in November 2015 via
https://github.com/rear/rear/pull/681

**Comment by jsmeix** [comment]
FYI
and for the fun of it some forensics:

According to
https://github.com/rear/rear/issues/544#issuecomment-120913214
the root is
https://bugzilla.opensuse.org/show_bug.cgi?id=220839
and according to its
https://bugzilla.opensuse.org/show_bug.cgi?id=220839#c42
the SUSE specific 'gpt_sync_mbr' partitioning scheme
was introduced in July 2008.

In November 2015 I implemented basic support for it in ReaR.

In March 2016 the SUSE patches for 'gpt_sync_mbr' were removed.

Now in February 2024 support for 'gpt_sync_mbr' in ReaR
will get shown as deprecated via ErrorIfDeprecated.

Let's wait and see how long it will take until the last system
with old inherited 'gpt_sync_mbr' partitioning has gone ;-)
